File: Python/GPU.py
from Komponen import Komponen

class GPU(Komponen):
    def __init__(self, vram_gb=0, chipset="", merk="", nama=""):
        super().__init__(merk, nama)
        self._vram_gb = vram_gb
        self._chipset = chipset

    def set_vram_gb(self, vram_gb):
        self._vram_gb = vram_gb

    def set_chipset(self, chipset):
        self._chipset = chipset

    def get_vram_gb(self):
        return self._vram_gb

    def get_chipset(self):
        return self._chipset

File: C++/GPU.cpp
#pragma once
#include <iostream>
#include <string>
#include "Komponen.cpp"

using namespace std;

class GPU : virtual public Komponen
{
private:
    int vramGB;
    string chipset;

public:
    GPU() {}

    GPU(int vramGB, string chipset, string merk, string nama) : Komponen(merk, nama)
    {
        this->vramGB = vramGB;
        this->chipset = chipset;
    }

    void setVramGB(int vramGB) { this->vramGB = vramGB; }
    void setChipset(string chipset) { this->chipset = chipset; }

    int getVramGB() { return vramGB; }
    string getChipset() { return chipset; }

    virtual ~GPU() {}
};
